### PR TITLE
grafana/data: Move useFieldOverrides from core

### PR DESCRIPTION
**Pull request by dprokop**
Moves [useFieldOverrides](https://github.com/grafana/grafana/blob/main/public/app/features/panel/components/PanelRenderer.tsx#L121-L122) hook from core to @grafana/data.

Additional changes
- API of the hook changed so that theme and interpolation function can be provided from outside
- react-use dependency was added to @grafana/data

This change is necessary for @grafana/scenes library.

### CODE REVIEW

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 1]
```diff
@@ -503,3 +509,51 @@ export function applyRawFieldOverrides(data: DataFrame[]): DataFrame[] {
...
+    return {
+      structureRev: structureRev.current,
+      ...data,
+      series: applyFieldOverrides({
+        data: series,
+        fieldConfig,
+        fieldConfigRegistry,
+        replaceVariables: (str: string, scopedVars?: ScopedVars) => {
+          return replace(str, scopedVars);
+        },
```
Comment (torkelo) [inline packages/grafana-data/src/field/fieldOverrides.ts:553]:
```suggestion
        replaceVariables: replace,
```

could we not just do this here?

Comment (dprokop) [reply packages/grafana-data/src/field/fieldOverrides.ts:553]: Of couse, my bad.

File: public/app/features/panel/components/PanelRenderer.tsx [hunk 2]
```diff
@@ -1,18 +1,13 @@
 import React, { useState, useMemo, useEffect, useRef } from 'react';
-import { usePrevious } from 'react-use';
```
Comment (Clarity-89) [inline public/app/features/panel/components/PanelRenderer.tsx:2]: Looks like `useRef` is not used here anymore.

Comment (dprokop) [reply public/app/features/panel/components/PanelRenderer.tsx:2]: @Clarity-89 thanks!

**Review by Clarity-89 — APPROVED** [review]

**Pull request merged**